tests/network/example_fuji_network.go

```diff
@@ -0,0 +1,146 @@
+	if err != nil {
+		panic(err)
+	}
+	conduitBlockchainID, err = ids.FromString("9asUA3QckLh7vGnFQiiUJGPTx8KE4nFtP8c1wTWJuP8XiWW75")
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Implements Network, pointing to subnets deployed on Fuji
+type FujiNetwork struct{}
+
+func (g *FujiNetwork) GetSubnetsInfo() []utils.SubnetTestInfo {
+	amplifyWSURI := "wss://subnets.avax.network/amplify/testnet/ws"
+	amplifyWSClient, err := ethclient.Dial(amplifyWSURI)
+	Expect(err).Should(BeNil())
+	amplifyRPCURI := "https://subnets.avax.network/amplify/testnet/rpc"
+	amplifyRPCClient, err := ethclient.Dial(amplifyRPCURI)
+	Expect(err).Should(BeNil())
+	amplifyChainIDInt, err := amplifyRPCClient.ChainID(context.Background())
+	Expect(err).Should(BeNil())
+
+	bulletinWSURI := "wss://subnets.avax.network/bulletin/testnet/ws"
+	bulletinWSClient, err := ethclient.Dial(bulletinWSURI)
+	Expect(err).Should(BeNil())
+	bulletinRPCURI := "https://subnets.avax.network/bulletin/testnet/rpc"
+	bulletinRPCClient, err := ethclient.Dial(bulletinRPCURI)
+	Expect(err).Should(BeNil())
+	bulletinChainIDInt, err := bulletinRPCClient.ChainID(context.Background())
+	Expect(err).Should(BeNil())
+
+	conduitWSURI := "wss://subnets.avax.network/conduit/testnet/ws"
+	conduitWSClient, err := ethclient.Dial(conduitWSURI)
+	Expect(err).Should(BeNil())
+	conduitRPCURI := "https://subnets.avax.network/conduit/testnet/rpc"
+	conduitRPCClient, err := ethclient.Dial(conduitRPCURI)
+	Expect(err).Should(BeNil())
+	conduitChainIDInt, err := conduitRPCClient.ChainID(context.Background())
+	Expect(err).Should(BeNil())
+
+	amplifyInfo := utils.SubnetTestInfo{
+		SubnetID:       amplifySubnetID,
+		BlockchainID:   amplifyBlockchainID,
+		ChainIDInt:     amplifyChainIDInt,
+		ChainWSClient:  amplifyWSClient,
+		ChainRPCClient: amplifyRPCClient,
+	}
+	bulletinInfo := utils.SubnetTestInfo{
+		SubnetID:       bulletinSubnetID,
+		BlockchainID:   bulletinBlockchainID,
+		ChainIDInt:     bulletinChainIDInt,
+		ChainWSClient:  bulletinWSClient,
+		ChainRPCClient: bulletinRPCClient,
+	}
+	conduitInfo := utils.SubnetTestInfo{
+		SubnetID:       conduitSubnetID,
+		BlockchainID:   conduitBlockchainID,
+		ChainIDInt:     conduitChainIDInt,
+		ChainWSClient:  conduitWSClient,
+		ChainRPCClient: conduitRPCClient,
+	}
+
+	return []utils.SubnetTestInfo{
+		amplifyInfo,
+		bulletinInfo,
+		conduitInfo,
+	}
+}
+
+func (g *FujiNetwork) GetTeleporterContractAddress() common.Address {
+	return teleporterContractAddress
+}
+
+func (g *FujiNetwork) GetFundedAccountInfo() (common.Address, *ecdsa.PrivateKey) {
+	key, err := crypto.HexToECDSA(skHex)
+	Expect(err).Should(BeNil())
+
+	return userAddress, key
+}
+
+func (g *FujiNetwork) RelayMessage(ctx context.Context,
+	sourceBlockHash common.Hash,
+	sourceBlockNumber *big.Int,
+	source utils.SubnetTestInfo,
+	destination utils.SubnetTestInfo) *types.Receipt {
+
+	// Rely on a separately deployed relayer to relay the message
```

Comment:
Should we implement that a check that the message has been successfully relayed within the given ctx timeout here, or is that best to live elsewhere?

Reply:
I think the check should be here. What if we changed the implementation for both local and fuji to subscribe to events on the destination chain and wait for the given message ID until some timeout instead of checking the success the of tx we send ourselves in the local case?

Reply:
I added a timeout to the Fuji `RelayMessage` implementation, but not the local network implementation. There isn't really anything asynchronous in the local network implementation that could fail quietly, so I don't think a timeout is necessary. We can always add that in later if we start to observe tests hanging at that point.